hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreFileInfo.java

```diff
@@ -163,22 +168,57 @@ private StoreFileInfo(final Configuration conf, final FileSystem fs, final FileS
       }
       LOG.trace("{} is a {} reference to {}", p, reference.getFileRegion(), referencePath);
     } else if (isHFile(p)) {
-      // HFile
-      if (fileStatus != null) {
-        this.createdTimestamp = fileStatus.getModificationTime();
-        this.size = fileStatus.getLen();
-      } else {
-        FileStatus fStatus = fs.getFileStatus(initialPath);
-        this.createdTimestamp = fStatus.getModificationTime();
-        this.size = fStatus.getLen();
-      }
+      // If a fileStatus, use it. It is pointed at the hfile, not at a link or reference.
+      this.cachedFileStatus = fileStatus != null? fileStatus: fs.getFileStatus(p);
```

Comment:
So only in case of this is an HFile (not a split ref file and link ), we will keep the cachedFileStatus.
Not sure in such cases what status it will be passing.   Should be status of the ref file or link file not the referred one or linked one right.. Checking deeply that we wont break anything in that regard with this patch.